spec/integration/ssl/key_spec.rb

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+
+require 'puppet/ssl/key'
+
+describe Puppet::SSL::Key do
+  include PuppetSpec::Files
+
+  # different UTF-8 widths
+  # 1-byte A
+  # 2-byte ۿ - http://www.fileformat.info/info/unicode/char/06ff/index.htm - 0xDB 0xBF / 219 191
+  # 3-byte ᚠ - http://www.fileformat.info/info/unicode/char/16A0/index.htm - 0xE1 0x9A 0xA0 / 225 154 160
+  # 4-byte ܎ - http://www.fileformat.info/info/unicode/char/2070E/index.htm - 0xF0 0xA0 0x9C 0x8E / 240 160 156 142
+  let (:mixed_utf8) { "A\u06FF\u16A0\u{2070E}" } # Aۿᚠ܎
+
+  before do
+    # Get a safe temporary file
+    dir = tmpdir('key_integration_testing')
+
+    Puppet.settings[:confdir] = dir
+    Puppet.settings[:vardir] = dir
+
+    # Puppet::SSL::Host.ca_location = :local
+    Puppet::SSL::Host.ca_location = :none
```

Comment:
I think this relates to these definitions: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/host.rb#L92-L102, which seem to imply if this setting varies if we're on a master or agent